integration/spark/src/main/scala/org/apache/spark/sql/CarbonContext.scala

```diff
@@ -208,4 +208,36 @@ object CarbonContext {
     }
 
   }
+
+  /**
+   * Requesting the extra executors other than the existing ones.
+   *
+   * @param sc                   sparkContext
+   * @param requiredExecutors         required number of executors to be requested
+   * @param localityAwareTasks   The number of pending tasks which is locality required
+   * @param hostToLocalTaskCount A map to store hostname with its possible task number running on it
+   * @return
+   */
+  final def ensureExecutors(sc: SparkContext,
```

Comment:
Do not add this function here, add it to DistributionUtil, like in #365. It will be reused for spark2 integration